File: app/src/main/java/com/fstyle/structure_android/data/source/UserRepository.java
package com.fstyle.structure_android.data.source;

import com.fstyle.structure_android.data.model.User;
import com.fstyle.structure_android.data.source.local.sqlite.UserLocalDataSource;
import com.fstyle.structure_android.data.source.remote.UserRemoteDataSource;
import io.reactivex.Observable;
import java.util.List;

/**
 * Created by le.quang.dao on 10/03/2017.
 */

public class UserRepository {

    private UserDataSource.LocalDataSource mLocalDataSource;
    private UserDataSource.RemoteDataSource mRemoteDataSource;

    public UserRepository(UserLocalDataSource localDataSource,
            UserRemoteDataSource remoteDataSource) {
        mLocalDataSource = localDataSource;
        mRemoteDataSource = remoteDataSource;
    }

    public Observable<List<User>> searchUsers(String keyword, int limit) {
        return mRemoteDataSource.searchUsers(keyword, limit);
    }
}

File: app/src/main/java/com/fstyle/structure_android/data/source/UserRepository.kt
package com.fstyle.structure_android.data.source

import com.fstyle.structure_android.data.model.User
import com.fstyle.structure_android.data.source.remote.UserRemoteDataSource
import io.reactivex.Single

/**
 * Created by le.quang.dao on 10/03/2017.
 */

interface UserRepository : UserDataSource.RemoteDataSource

class UserRepositoryImpl(val remoteDataSource: UserRemoteDataSource) : UserRepository {

  override fun searchUsers(keyword: String, limit: Int): Single<List<User>> {
    return remoteDataSource.searchUsers(keyword, limit)
  }

  override fun getUserDetailFromServer(userLogin: String?): Single<User> {
    return remoteDataSource.getUserDetailFromServer(userLogin)
  }
}
